system/include/emscripten/bind.h

```diff
@@ -1419,10 +1495,8 @@ namespace emscripten {
         };
     }
 
-    template<typename K, typename V>
-    class_<std::map<K, V>> register_map(const char* name) {
-        typedef std::map<K,V> MapType;
-
+    template<typename K, typename V, typename MapType=std::map<K,V>>
+    class_<MapType> register_map(const char* name) {
```

Comment:
Why did you generalize this?  It may preclude us from binding other methods of std::map without breaking backwards compatibility.  (Someone may bind a different map type that does not bind all the methods of std::map.)